### PR TITLE
make Unix.sleepf available under all OCaml versions

### PR DESCRIPTION
**Pull request by UnixJunkie**

### CODE REVIEW

**Comment by UnixJunkie** [comment]
related to https://github.com/ocaml-batteries-team/batteries-included/issues/929

File: src/batUnix.mlv [hunk 1]
```diff
@@ -30,6 +30,27 @@ include Unix
 ##V<4.2##let send_substring = send
 ##V<4.2##let sendto_substring = sendto
 
+##V<4.3##let sleepf (timeout: float): unit =
+##V<4.3##  let _, _, _ = select [] [] [] timeout in
```
Comment (rixed) [inline src/batUnix.mlv]: You probably want to restart_on_EINTR.

Also, have you considered `Thread.delay` ?

Comment (UnixJunkie) [reply src/batUnix.mlv]: I'll have a look at that. Thanks for the review.

Comment (UnixJunkie) [reply src/batUnix.mlv]: Concerning thread.delay, it is implemented as Unix.sleepf in recent OCaml versions.

Comment (UnixJunkie) [reply src/batUnix.mlv]: the implementation became a little bit more complex

**Comment by UnixJunkie** [comment]
I also updated the changelog.

File: src/batUnix.mlv (hunk 1 above)
Comment (rixed) [reply src/batUnix.mlv]: I'm assuming `Thread.delay` does not use `Unix.sleepf` when V<4.3, so wouldn't that be easier and more reliable to just call `Thread.delay` then? Unless we do not want to link the Unix to the Thread module, maybe?

**Comment by UnixJunkie** [comment]
If you have a better PR, or if you can improve this one, please do it.

File: src/batUnix.mlv (hunk 1 above)
Comment (UnixJunkie) [reply src/batUnix.mlv]: They use thread_delay from scheduler.c, which calls schedule_thread, which looks like a real mess.
But, is the semantics of delaying a thread the same as making a process sleep?
I think a process could have several threads, in theory.

**Pull request merged**